### ISSUE TITLE
False positive when assigning function with positional-only separator

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#2511]
This is a tracking issue. For original see https://github.com/microsoft/pylance-release/issues/2006.

Here's a minimal repro:

```python
from typing import Any, Protocol

def f1(x: int, /, y: str, z: None = None) -> Any:
    ...

class X(Protocol):
    def __call__(self, x: int, /, y: str) -> Any:
        ...

x: X = f1
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.183, which I just published. It will also be included in the next release of pylance.
